src/Files/WeeklyFilePattern.php

```diff
@@ -32,7 +32,7 @@ public static function fromInfo(array $info, \DateTimeInterface $now = NULL) {
     $interval = new \DateInterval('P1W');
 
     $end = $now->modify('+1 day')->modify('last monday')->setTime(0, 0, 0, 0);
-    $start = $end->sub($info['retention_period']);
+    $start = $now->sub($info['retention_period'])->modify('-1 day')->modify('next monday')->setTime(0, 0, 0, 0);
```

Comment:
This also means having less data but everything is within the retention period instead of all data from the retention period plus maybe some older data? :thinking:

Reply:
Yep. I see the former behaviour as a bug. It would still generate/keep files that are older than the retention period.